src/main.rs

```diff
@@ -0,0 +1,254 @@
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_2: bool,
+
+    #[clap(
+        name = "level_3",
+        short = '3',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_3: bool,
+
+    #[clap(
+        name = "level_4",
+        short = '4',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_4: bool,
+
+    #[clap(
+        name = "level_5",
+        short = '5',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_5: bool,
+
+    #[clap(
+        name = "level_6",
+        short = '6',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_6: bool,
+
+    #[clap(
+        name = "level_7",
+        short = '7',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_7: bool,
+
+    #[clap(
+        name = "level_8",
+        short = '8',
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+        hide = true
+    )]
+    level_8: bool,
+
+    /// Compress better
+    #[clap(
+        name = "level_9",
+        short = '9',
+        long = "best",
+        multiple_occurrences = true,
+        overrides_with_all = &[
+            "level_1",
+            "level_2",
+            "level_3",
+            "level_4",
+            "level_5",
+            "level_6",
+            "level_7",
+            "level_8",
+            "level_9"
+        ],
+    )]
+    level_9: bool,
```

Comment:
I liked the idea of having a separate integral `compression_level` field populated from the boolean flags (or the default). Hopefully not too much work to make one?

Reply:
While I was tinkering, the way I had it set was as a function `Args::compression_level(&self) -> u32`.